salt/beacons/cert_info.py

```diff
@@ -100,11 +100,11 @@ def beacon(config):
 
         if isinstance(cert_path, dict):
             try:
-                notify_days = cert_path[cert_path.keys()[0]].get(
+                notify_days = cert_path[next(iter(cert_path.keys()))].get(
                     "notify_days", global_notify_days
                 )
-                cert_path = cert_path.keys()[0]
-            except IndexError as exc:
+                cert_path = next(iter(cert_path.keys()))
```

Comment:
Call duplication. It's better to move this line upper and re-use `cert_path` value in the call above.

Reply:
Its duplicated but if we move it above then the indexed get of `cert_path` on line 103 would be on the next item, not the current. I think in it's current state this is probably supposed to be:
```

            try:
                notify_days = cert_path[next(iter(cert_path))].get(
                    "notify_days", global_notify_days
                )
                
            except StopIteration as exc:
                log.error("Unable to load certificate %s (%s)", cert_path, exc)
                continue
            else:
                cert_path = next(iter(cert_path))
```
So I'm going to update with:
```
            try:
                next_cert_path = next(iter(cert_path))
                notify_days = cert_path[next_cert_path].get(
                    "notify_days", global_notify_days
                )
            except StopIteration as exc:
                log.error("Unable to load certificate %s (%s)", cert_path, exc)
                continue
            else:
                cert_path = next_cert_path
```
if that seems ok

Reply:
Actually, that's probably more correct as
```
            try:
                next_cert_path = next(iter(cert_path))
            except StopIteration as exc:
                log.error("Unable to load certificate %s (%s)", cert_path, exc)
                continue
            else:
                notify_days = cert_path[next_cert_path].get(
                    "notify_days", global_notify_days
                )
                cert_path = next_cert_path
```